Userland/Libraries/LibWeb/DOM/ElementFactory.cpp

```diff
@@ -291,8 +292,10 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<Element>> create_element(Document& document
     if (lowercase_tag_name == SVG::TagNames::text)
         return MUST_OR_THROW_OOM(realm.heap().allocate<SVG::SVGTextContentElement>(realm, document, move(qualified_name)));
 
-    // FIXME: If name is a valid custom element name, then return HTMLElement.
-
+    // If name is a valid custom element name, then return HTMLElement.
+    if (HTML::is_valid_custom_element_name(TRY_OR_THROW_OOM(document.vm(), String::from_deprecated_string(local_name)))) {
```

Comment:
The point of making `is_valid_custom_element_name` take a `StringView` is that you don't need to do this allocation to `String`. `StringView` accepts both `String` and `DeprecatedString`.